src/lib/validators.ts

```diff
@@ -0,0 +1,138 @@
+'use strict';
+import * as Bluebird from 'bluebird';
+import { stat, statSync } from 'fs';
+const statAsync = Bluebird.promisify(stat);
+
+export async function validateDirectoryPath(
+	path: string,
+	raise = true,
+): Promise<boolean> {
+	try {
+		const stat = await statAsync(path);
+		if (!stat.isDirectory()) {
+			if (raise) {
+				throw new Error('Error: ' + path + ' is not a directory');
+			}
+			return false;
+		}
+		return true;
+	} catch (error) {
+		if (raise) {
+			throw error;
+		}
+		return false;
+	}
+}
+
+export function validateDirectoryPathSync(path: string, raise = true): boolean {
+	try {
+		const st = statSync(path);
+		if (!st.isDirectory()) {
+			if (raise) {
+				throw new Error('Error: ' + path + ' is not a directory');
+			}
+			return false;
+		}
+		return true;
+	} catch (error) {
+		if (raise) {
+			throw error;
+		}
+		return false;
+	}
+}
+
+export async function validateFilePath(
+	path: string,
+	raise = true,
+): Promise<boolean> {
+	try {
+		const stat = await statAsync(path);
+		if (!stat.isFile()) {
+			if (raise) {
+				throw new Error('Error: ' + path + ' is not a file');
+			}
+			return false;
+		}
+		return true;
+	} catch (error) {
+		if (raise) {
+			throw error;
+		}
+		return false;
+	}
+}
+
+export function validateFilePathSync(path: string, raise = true): boolean {
+	try {
+		const stat = statSync(path);
+		if (!stat.isFile()) {
+			if (raise) {
+				throw new Error('Error: ' + path + ' is not a file');
+			}
+			return false;
+		}
+		return true;
+	} catch (error) {
+		if (raise) {
+			throw error;
+		}
+		return false;
+	}
+}
+
+export function validateFQDN(value: any, raise = true): boolean {
```

Comment:
There's a great NPM module, https://www.npmjs.com/package/validator , which does a load of these which have been tested/verified already.

Reply:
Added and using `isFQDN` and `isPort` validators. We might want to treat the domain `kubernetes` as valid (it's invalid for isFQDN).